synapse/crypto/keyclient.py

```diff
@@ -82,6 +82,12 @@ def connectionMade(self):
         self._peer = self.transport.getPeer()
         logger.debug("Connected to %s", self._peer)
 
+        if not isinstance(self.path, bytes):
+            self.path = self.path.encode('ascii')
```

Comment:
Can paths and hosts not be unicode?

Reply:
nope, must be ascii-encoded (maybe punycode if it's a hostname or url-encoded if it's a path), because http